_posts/2024-03-20-infinispan-15-tracing.adoc

```diff
@@ -0,0 +1,135 @@
+In the demo this is the configuration passed to the Infinispan Helm Chart:
+
+[source,yaml]
+----
+infinispan:
+ cacheContainer:
+   tracing:
+     collector-endpoint: "http://jaeger-collector.infinispan-15-tracing.svc.cluster.local:4318"
+     service-name: infinispan-service-tracing
+     security: true
+----
+
+=== Cache configuration
+
+The other tracing categories are configurable at cache level, since they are cache scoped.
+For instance this is a possible cache configuration enabling the tracing on more categories:
+
+[source,yaml]
+----
+replicatedCache:
+  encoding:
+    key:
+      mediaType: "application/x-protostream"
+    value:
+      mediaType: "application/x-protostream"
+  indexing:
+    enabled: "true"
+    storage: "local-heap"
+    startupMode: "NONE"
+    indexedEntities:
+      - "fax.play.image"
+  tracing:
+    enabled: true
+    categories:
+      - "container"
+      - "persistence"
+      - "cluster"
+      - "x-site"
+----
+
+In this case the enabled tracing categories will be container, persistence, cluster and x-site.
+By default only the container category is enabled.
+
+==== Runtime enabling / disabling them
+
+Furthermore, it is possible to enable / disable them or even the entire tracing at runtime.
+
+For instance this is a piece of code using the HotRod client to enable at runtime the category `container` and `persistence`,
+disabling the others. A similar command can be invoked also with the REST interface or the command line interface.
+
+[source,java]
+----
+Set<String> enabledCategories = Set.of("container", "persistence");
+cacheManager.administration()
+   .updateConfigurationAttribute(cache.getName(), "tracing.categories", String.join(",", enabledCategories));
+----
+
+== Tracing in action
+
+Now that we have the setup. Let's see some example of tracing in action.
+If we open the Jaeger console we can see immediately some security audit events logged.
+
+[caption="Security audit",link=/assets/images/blog/tracing-15-01-security-audit.png]
+image::/assets/images/blog/tracing-15-01-security-audit.png[Tracing output]
+
+When the demo starts we only have enabled the container tracing.
+So after the first interaction we can have a log similar to the following:
+
+[caption="Security audit",link=/assets/images/blog/tracing-15-02-container.png]
+image::/assets/images/blog/tracing-15-02-container.png[Tracing output]
+
+We have the span from the client and the corresponding span of container level of the server
+(that traces the HotRod call).
+
+If we want to know more about the internal Infinispan operations we need to enable more span categories.
+
+In our demo for instance we first enable (at runtime) the persistence category
+and after another interaction from the client we client we can see also the internal call that Infinispan does
+to interact with the configured cache store.
```

Comment:
In our demo we initially enable the persistence category at runtime. After some interactions from the client, we can observe the internal calls that Infinispan makes to interact with the configured cache store.